### ISSUE TITLE
Premium block flows: save revision before leaving the editor

### ISSUE THREAD

**Issue by simison** [Automattic/jetpack#12976]
TL;DR: instead of showing _"you have unsaved modifications"_ when pressing "upgrade", save a revision and return to that revision after checkout.

### Current behavior

Currently, when the customer presses "ugprade", we redirect them out of the editor. Their post might be new and unsaved, draft, already published, containing changes or not.

Block editor/browser notices unsaved modifications and warns with a popup:

![image](https://user-images.githubusercontent.com/87168/60790387-0f7d8380-a16a-11e9-9c15-8407f2024f8d.png)

That can be scary.

### Ideal behavior
Instead, we should dispatch an autosave to save a new revision of the post and return to that revision after checkout without customer's post being affected but allowing them to continue where they left off.

### Notes

You can test autosave behaviour with [`savePost()`](https://github.com/WordPress/gutenberg/blob/a8e96920032ab4f9ccb38c29de34f90c296ed7a7/packages/editor/src/store/actions.js#L258):
```
wp.data.dispatch( 'core/editor' ).savePost({ isAutosave: true });
```

Or with [`autosave()`](https://github.com/WordPress/gutenberg/blob/a8e96920032ab4f9ccb38c29de34f90c296ed7a7/packages/editor/src/store/actions.js#L513):
```
wp.data.dispatch( 'core/editor' ).autosave();
```

If the customer just opens the page without returning to specific autosave, they'd see this:

<img width="793" alt="Screenshot 2019-07-08 at 10 22 30" src="https://user-images.githubusercontent.com/87168/60791258-e827b600-a16b-11e9-84d5-30c2c4ab85ec.png">

And the wp-admin URL would be `/wp-admin/post.php?post=20&action=edit&revision=25`. I'm not sure if we have such URL in Calypso block editor since revisions are loaded via modal like this:

<img width="706" alt="Screenshot 2019-07-08 at 10 34 50" src="https://user-images.githubusercontent.com/87168/60791339-1c9b7200-a16c-11e9-98fd-f9425f1947b5.png">

[There's some existing related code](https://github.com/Automattic/wp-calypso/blob/49b509b138067d70b7d2c738f281b8174956d468/apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js#L102-L125) to bridge Calypso and Gutenberg here, that might be helpful:

This shouldn't be necessary for 1st iteration but something we could investigate little further.

cc @mtias @Automattic/luna

**Comment by mtias** [comment]
Yes, saving before leaving seems perfectly fine. Longer term, I'd want us to explore a flow that does the checkout within a modal so you never leave the editor (at least in Calypso).
